src/main/java/com/dongxuexidu/douban4j/provider/OAuthDoubanProvider.java

```diff
@@ -148,7 +148,7 @@ public AccessToken tradeAccessTokenWithCode (String code) throws DoubanException
       Map<String,String> params = new HashMap<String, String>();
       params.put("client_id", this.apiKey);
       params.put("client_secret", this.secretKey);
-      params.put("redirect_uri", DefaultConfigs.ACCESS_TOKEN_REDIRECT_URL);
+      params.put("redirect_uri",  this.redirectUrl);
```

Comment:
Guess you need to check if it's null first, and if it's null, set it back to the default one.